rdt/transformers/null.py

```diff
@@ -81,22 +75,19 @@ def transform(self, data):
         Returns:
             numpy.ndarray
         """
-        if self.nulls:
+        if self._null_column or (self.nulls and self._fill_value is not None):
+            isna = data.isna()
             if not isinstance(data, pd.Series):
                 data = pd.Series(data)
 
-            isna = data.isna()
-            if self.nulls and self._fill_value is not None:
-                if not self.copy:
-                    data[isna] = self._fill_value
-                else:
-                    data = data.fillna(self._fill_value)
-
-            if self._null_column:
-                return pd.concat([data, isna.astype('int')], axis=1).to_numpy()
+        if self.nulls and self._fill_value is not None:
+            if not self.copy:
+                data[isna] = self._fill_value
+            else:
+                data = data.fillna(self._fill_value)
 
-            if self._fill_value in data:
-                warnings.warn(IRREVERSIBLE_WARNING)
```

Comment:
@csala So apparently, generating this warning takes the most time out of the whole process (fitting, transforming, reverse transforming)... if the data has lots of rows with nan's, over 90% of the time is spent generating this warning... for now I just removed the warning altogether, so it passes the test cases.